Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -24,6 +30,132 @@ namespace AzToolsFramework
+                if (path == m_viewBookmarksKey && localBookmarksId.has_value() && !localBookmarksId->empty())
+                {
+                    if (const auto existingBookmarkEntry = m_bookmarkMap.find(localBookmarksId.value());
+                        existingBookmarkEntry == m_bookmarkMap.end())
+                    {
+                        m_bookmarkMap.insert(AZStd::make_pair(localBookmarksId.value(), AZStd::vector<ViewBookmark>{}));
+                    }
+                }
+            }
+
+            return AZ::SettingsRegistryInterface::VisitResponse::Continue;
+        }
+
+        using AZ::SettingsRegistryInterface::Visitor::Visit;
+        void Visit(AZStd::string_view path, AZStd::string_view valueIndex, AZ::SettingsRegistryInterface::Type, double value) override
+        {
+            AZ::StringFunc::TokenizeLast(path, "/");
+            AZStd::optional<AZStd::string_view> dataType = AZ::StringFunc::TokenizeLast(path, "/");
+            AZStd::optional<AZStd::string_view> bookmarkIndexStr = AZ::StringFunc::TokenizeLast(path, "/");
+            AZStd::optional<AZStd::string_view> bookmarkType;
+            if (bookmarkIndexStr == LastKnownLocationKey)
+            {
+                bookmarkType = bookmarkIndexStr;
+            }
+            else
+            {
+                // differentiate between local Bookmarks and LastKnownLocation
+                bookmarkType = AZ::StringFunc::TokenizeLast(path, "/");
+            }
+
+            if (AZStd::optional<AZStd::string_view> localBookmarksId = AZ::StringFunc::TokenizeLast(path, "/");
+                localBookmarksId.has_value() && path == m_viewBookmarksKey && !localBookmarksId->empty())
+            {
+                auto setVec3Fn = [value](AZ::Vector3& inout, int currentIndex)
+                {
+                    switch (currentIndex)
+                    {
+                    case 0:
+                        inout.SetX(aznumeric_cast<float>(value));
+                        break;
+                    case 1:
+                        inout.SetY(aznumeric_cast<float>(value));
+                        break;
+                    case 2:
+                        inout.SetZ(aznumeric_cast<float>(value));
+                        break;
+                    default:
+                        AZ_Warning(
+                            "LocalViewBookmarkLoader", false, "Trying to set an invalid index in a Vector3, index = %d", currentIndex);
+                        break;
+                    }
+                };
+
+                if (bookmarkType == LastKnownLocationKey)
+                {
+                    if (!m_lastKnownLocation.has_value())
+                    {
+                        m_lastKnownLocation.emplace(ViewBookmark{});
+                    }
+
+                    const int currentIndex = AZStd::stoi(AZStd::string(valueIndex));
+                    if (dataType == "Position")
```

Comment:
`dataType` is an optional, shouldn't all these string comparisons be `if (*dataType == "Position")`?

Otherwise is it under the hood creating an `optional<string>` out of `"Position"` and then comparing the optionals?

Reply:
Hmm you could be right but this code is just moved from before and I'd like to rewrite some of this in future anyway so I'm inclined to leave this as-is for now (I'll have a look into it to check tomorrow though)